src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingSourceGenerator.Emitter.cs

```diff
@@ -0,0 +1,668 @@
+                    EmitBindLogicFromIConfiguration(type, Literal.obj, InitializationKind.Declaration);
+                    _writer.WriteLine($"return (T)({GlobalName.Object}){Literal.obj};");
+                    _writer.WriteBlockEnd();
+                    _writer.WriteBlankLine();
+                }
+
+                Emit_NotSupportedException_UnableToBindType(NotSupportedReason.TypeNotDetectedAsInput);
+                _writer.WriteBlockEnd();
+                _writer.WriteBlankLine();
+            }
+
+            private void EmitBindMethods()
+            {
+                if (_generationSpec.TypesForBindMethodGen.Count > 0)
+                {
+                    foreach (TypeSpec type in _generationSpec.TypesForBindMethodGen)
+                    {
+                        EmitBindMethod(type);
+                    }
+                }
+
+                // Get & Configure method generation might have queued types for private BindCore impl
+                while (_privateBindCoreMethodGen_QueuedTypes.Count > 0)
+                {
+                    EmitBindCoreMethod(_privateBindCoreMethodGen_QueuedTypes.Dequeue());
+                }
+            }
+
+            private void EmitBindMethod(TypeSpec type)
+            {
+                if (_internalBindMethodGen_ProcessedTypes.Contains(type))
+                {
+                    return;
+                }
+
+                _internalBindMethodGen_ProcessedTypes.Add(type);
+
+                // Binding to root level struct is a no-op.
+                // TODO: maybe this should be a debug assert & the parser shouldn't include them.
+                if (type.IsValueType)
+                {
+                    return;
+                }
+
+                _privateBindCoreMethodGen_QueuedTypes.Enqueue(type);
+
+                _writer.WriteLine(
+                    @$"public static void {Literal.Bind}(this {GlobalName.IConfiguration} {Literal.configuration}, {type.DisplayString} {Literal.obj}) => " +
+                        $"{Literal.BindCore}({Literal.configuration}, ref {Literal.obj});");
+                _writer.WriteBlankLine();
+            }
+
+            private void EmitBindCoreMethod(TypeSpec type)
+            {
+                if (_privateBindCoreMethodGen_ProcessedTypes.Contains(type))
+                {
+                    return;
+                }
+                _privateBindCoreMethodGen_ProcessedTypes.Add(type);
+
+                string objParameterExpression = $"ref {type.DisplayString} {Literal.obj}";
+                _writer.WriteBlockStart(@$"private static void {Literal.BindCore}({GlobalName.IConfiguration} {Literal.configuration}, {objParameterExpression})");
+                EmitBindCoreImpl(type);
+                _writer.WriteBlockEnd();
+                _writer.WriteBlankLine();
+            }
+
+            private void EmitBindCoreImpl(TypeSpec type)
```

Comment:
>EmitBindCoreImpl [](http://example.com/codeflow?start=25&length=16)

Did you write somewhere what we'll support in this PR and what we support later?